File: leetcode/lc-us/cpp/191.cpp
#include <bits/stdc++.h>

class Solution {
public:
    int hammingWeight(uint32_t n) {
        int hamming_weight = 0;
        for (int i = 0; i < 32; i++) {
            if ((n & 1) == 1) {
                hamming_weight++;
            }
            n >>= 1;
        }
        return hamming_weight;
    }
};

File: leetcode/lc-us/rust/191.rs
impl Solution {
    pub fn hammingWeight(n: u32) -> i32 {
        let mut n = n;
        let mut res = 0;
        while n != 0 {
            if (n & 1) == 1 {
                res += 1;
            }
            n >>= 1;
        }
        res
    }
}